src/api/java/com/minecolonies/api/tileentities/TileEntityRack.java

```diff
@@ -133,7 +133,8 @@ public int getFreeSlots()
     @Override
     public boolean hasItemStack(final ItemStack stack, final boolean ignoreDamageValue)
     {
-        return content.get(new ItemStorage(stack, ignoreDamageValue)) != null;
+        final ItemStorage checkItem = new ItemStorage(stack, ignoreDamageValue);
+        return content.containsKey(checkItem) && content.get(checkItem) >= stack.getCount(); 
```

Comment:
don't you need content.get(checkItem).getAmount() ?

Reply:
@someaddons content.get(checkItem) returns and integer which is quantity. So, no, you don't need getAmount. 

@OrionDevelopment this is core TileEntityRack behavior that I'm just restoring previous behavior to. The flags are applied at another layer, and have never been driven down to here.

Reply:
Okey.

Reply:
Okey.